### PR TITLE
Pillview

### PR DESCRIPTION
**Pull request by danthecodingman**
Category horizontal scroll pills header nib

### CODE REVIEW

File: fbuapp/Models/Category.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface Category : PFObject <PFSubclassing>
+
+@property (nonatomic, strong) NSString *eventCategory;
+@property (nonatomic, strong) NSArray *eventCategoryColor;
+@property (nonatomic, assign) NSInteger eventCategoryID;
+
++ (void) addCategory: (NSString *)category withID: (NSInteger) categoryID withColor: (NSArray *) categoryColor withCompletion: (PFBooleanResultBlock  _Nullable)completion;
```
Comment (NSDavidObject) [inline fbuapp/Models/Category.h]:
```suggestion
+ (void)addCategory:(NSString *)category id:(NSInteger)categoryID color:(NSArray *)color completion:(PFBooleanResultBlock  _Nullable)completion;
```

Comment (NSDavidObject) [reply fbuapp/Models/Category.h]: You shouldn't need this function at all actually, why would the iOS app have the capability of adding new categories? It should be server only

File: fbuapp/Models/Category.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+#import <Parse/Parse.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface Category : PFObject <PFSubclassing>
+
+@property (nonatomic, strong) NSString *eventCategory;
+@property (nonatomic, strong) NSArray *eventCategoryColor;
+@property (nonatomic, assign) NSInteger eventCategoryID;
```
Comment (NSDavidObject) [inline fbuapp/Models/Category.h]: Shouldn't the class be named "EventCategory", and the properties be "color"/"id"/"name"?

File: fbuapp/Views/CategoryHeaderView.m [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+- (NSInteger)collectionView:(nonnull UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
+    return self.categories.count;
+}
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath{
+    PillCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:kCollectionViewPillCell forIndexPath:indexPath];
+    Category* category = self.categories[indexPath.row];
+    
+    NSArray *colorRGB = category.eventCategoryColor;
+    cell.pillBackground.backgroundColor = [UIColor colorWithRed: [colorRGB[0] doubleValue]/255.0 green:[colorRGB[1] doubleValue]/255.0 blue:[colorRGB[2] doubleValue]/255.0 alpha:1.0];
```
Comment (NSDavidObject) [inline fbuapp/Views/CategoryHeaderView.m]: Couldn't this be a UIColor category helper?

File: fbuapp/Views/CategoryHeaderView.m [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    cell.eventCategory.text = category.eventCategory;
+    cell.pillBackground.layer.cornerRadius = 17;
```
Comment (NSDavidObject) [inline fbuapp/Views/CategoryHeaderView.m]: Can we compute the corner radius value instead of hardcoding it?

**Review by NSDavidObject — CHANGES_REQUESTED** [review]

**Review by eazheng — APPROVED** [review]

File: fbuapp/View Controllers/HomeViewController.m [hunk 5]
```diff
@@ -13,6 +13,8 @@
 #import <CoreLocation/CoreLocation.h>
 #import "UIImageView+AFNetworking.h"
 #import "DateTools.h"
+#import "CategoryHeaderView.h"
+#import "EventCategory.h"
```
Comment (eazheng) [inline fbuapp/View Controllers/HomeViewController.m]: not needed import

**Pull request merged**